linux/LinuxProcessList.c

```diff
@@ -1193,9 +1175,16 @@ static bool LinuxProcessList_readCmdlineFile(Process* process, const char* dirna
    if (fd != -1)
       close(fd);
 
+   char filename[MAX_NAME + 1];
+
    /* execve could change /proc/[pid]/exe, so procExe should be updated */
-   xSnprintf(command, sizeof(command), "%s/%s/exe", dirname, name);
-   if ((amtRead = readlink(command, filename, sizeof(filename) - 1)) > 0) {
+#if defined(HAVE_READLINKAT) && defined(HAVE_OPENAT)
+   if ((amtRead = readlinkat(procFd, "exe", filename, sizeof(filename) - 1)) > 0) {
+#else
+   char path[4096];
+   xSnprintf(path, sizeof(path), "%s/exe", procFd);
+   if ((amtRead = readlink(path, filename, sizeof(filename) - 1)) > 0) {
+#endif
```

Comment:
I'd prefer if we could avoid starting a block within conditionally compiled sections of code.
```suggestion
#if defined(HAVE_READLINKAT) && defined(HAVE_OPENAT)
   amtRead = readlinkat(procFd, "exe", filename, sizeof(filename) - 1);
#else
   char path[4096];
   xSnprintf(path, sizeof(path), "%s/exe", procFd);
   amtRead = readlink(path, filename, sizeof(filename) - 1);
#endif
   if (amtRead > 0) {
```